lib/mac/bindings.js

```diff
@@ -184,6 +184,69 @@ NobleBindings.prototype.discoverDescriptors = function(identifier, serviceIdenti
   characteristic.discoverDescriptors();
 };
 
+NobleBindings.prototype.read = function(identifier, serviceIdentifier, characteristicIdentifier) {
+  var characteristic = this._characteristicForIdentifier(identifier, serviceIdentifier, characteristicIdentifier);
+
+  characteristic.readValue();
+
+  //can we get isNotification from corebluetooth?
+  characteristic.once('valueUpdate', function(value, error) {
+    this.emit('read', identifier, serviceIdentifier, characteristicIdentifier, value, null);
+  }.bind(this));
+};
+
+NobleBindings.prototype.write = function(identifier, serviceIdentifier, characteristicIdentifier, data, withoutResponse) {
+  var characteristic = this._characteristicForIdentifier(identifier, serviceIdentifier, characteristicIdentifier);
+
+  // TODO convert type here or in core-bluetooth peripheral?
+  characteristic.writeValue(data, type);
+
+  //TODO we do emit for withoutResponse right?
+  if (withoutResponse) {
+    this.emit('write', identifier, serviceIdentifier, characteristicIdentifier);
+  } else {
+    characteristic.once('valueWrite', function(error) {
+      this.emit('write', identifier, serviceIdentifier, characteristicIdentifier);
+    }.bind(this));
+  }
+};
+
+NobleBindings.prototype.readValue = function(identifier, serviceIdentifier, characteristicIdentifier, descriptorIdentifier) {
+  var descriptor = this._descriptorForIdentifier(identifier, serviceIdentifier, characteristicIdentifier, descriptorIdentifier);
+
+  descriptor.readValue();
+
+  // TODO this emits the handle right? but we dont have access to those anymore right?
```

Comment:
`descriptor.identifier` is the handle, so we can emit if needed.

However, since node-code-bluetooth doesn't support `readHandle` and `writeHandle` I don't think we should.